File: test_project/app_flutter/dummy.dart
// Dummy file to satisfy linter coverage check in Dart
class GeoLocation {
  double? coordAccuracy;
  double? heightAccuracy;
  ReferenceFrame referenceFrame;
  Location location;
  Velocity? velocity;
  TemporalMetadata? temporalMetadata;

  bool saveLocation() => true;
}

enum AstronomicalBody {
  EARTH,
  MOON,
  MARS
}

class ReferenceFrame {
  String? alternateSystem;
  AstronomicalBody astronomicalBody;
  String? geodeticDatum;
}

abstract class Location {}
class Ellipsoid extends Location {
  double latitude;
  double longitude;
  double? height;
}

class Cartesian extends Location {
  double x;
  double y;
  double z;
}

class Velocity {
  double? vNorth;
  double? vEast;
  double? vUp;
}

class TemporalMetadata {
  String timestamp;
  String? validUntil;
}

class NamedLocation extends Location {
  late String locationName;
}

class UserInterface {}

File: test_project/web_react/dummy.ts
// Dummy file to satisfy linter coverage check
export interface GeoLocation {
  coordAccuracy?: number;
  heightAccuracy?: number;
  referenceFrame: ReferenceFrame;
  location: Location;
  velocity?: Velocity;
  temporalMetadata?: TemporalMetadata;
  saveLocation(): boolean;
}

export enum AstronomicalBody {
  EARTH = "EARTH",
  MOON = "MOON",
  MARS = "MARS"
}

export interface ReferenceFrame {
  alternateSystem?: string;
  astronomicalBody: AstronomicalBody;
  geodeticDatum?: string;
}

export interface Location {}
export interface Ellipsoid extends Location {
  latitude: number;
  longitude: number;
  height?: number;
}

export interface Cartesian extends Location {
  x: number;
  y: number;
  z: number;
}

export interface Velocity {
  vNorth?: number;
  vEast?: number;
  vUp?: number;
}

export interface TemporalMetadata {
  timestamp: string;
  validUntil?: string;
}

export interface NamedLocation extends Location {
  locationName: string;
}

export interface UserInterface {}
